### ISSUE TITLE
macOS tests fail

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#39]
When running the CIs, it was just observed that the macOS CIs failed.

Below is the `pytest` verbose as observed [here](https://github.com/andreped/GradientAccumulator/actions/runs/4629533930/jobs/8189920947):
```
ImportError while importing test module '/Users/runner/work/GradientAccumulator/GradientAccumulator/tests/test_train_mnist.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../hostedtoolcache/Python/3.8.14/x64/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_train_mnist.py:2: in <module>
    import tensorflow_datasets as tfds
../../../hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/tensorflow_datasets/__init__.py:43: in <module>
    import tensorflow_datasets.core.logging as _tfds_logging
../../../hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/tensorflow_datasets/core/__init__.py:[22](https://github.com/andreped/GradientAccumulator/actions/runs/4629533930/jobs/8189920947#step:8:23): in <module>
    from tensorflow_datasets.core import community
../../../hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/tensorflow_datasets/core/community/__init__.py:18: in <module>
    from tensorflow_datasets.core.community.huggingface_wrapper import mock_builtin_to_use_gfile
../../../hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/tensorflow_datasets/core/community/huggingface_wrapper.py:31: in <module>
    from tensorflow_datasets.core import dataset_builder
../../../hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/tensorflow_datasets/core/dataset_builder.py:34: in <module>
    from tensorflow_datasets.core import dataset_info
../../../hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/tensorflow_datasets/core/dataset_info.py:47: in <module>
    from tensorflow_datasets.core import file_adapters
../../../hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/tensorflow_datasets/core/file_adapters.py:[29](https://github.com/andreped/GradientAccumulator/actions/runs/4629533930/jobs/8189920947#step:8:30): in <module>
    from array_record.python import array_record_module
E   ImportError: dlopen(/Users/runner/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/array_record/python/array_record_module.so, 2): no suitable image found.  Did find:
E   	/Users/runner/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/array_record/python/array_record_module.so: unknown file type, first eight bytes: 0x7F 0x45 0x4C 0x46 0x02 0x01 0x01 0x03
E   	/Users/runner/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/site-packages/array_record/python/array_record_module.so: unknown file type, first eight bytes: 0x7F 0x45 0x4C 0x46 0x02 0x01 0x01 0x03
```

**Comment by andreped** [comment]
This was the corresponding tf-datasets version from the verbose:
```
Collecting tensorflow-datasets
  Downloading tensorflow_datasets-4.9.0-py3-none-any.whl (5.4 MB)
```

We should likely try using an older tf-datasets version.

Apparently, this release was made yesterday (see [here](https://github.com/tensorflow/datasets/releases/tag/v4.9.0)).

**Comment by andreped** [comment]
Fixed in 3a2eb5b46bb2a95dd29895e6ca494ab2299fca12.

It was related to numerous things:
1. Batch norm layer was implemented wrongly according to the TF modern standards
2. TF-addons installation had no versioning - newer versions of addions were not compatible with older versions of TF
3. Same as above but for TF-datasets.
